### PR TITLE
Add tx_state_change event

### PR DESCRIPTION
**Pull request by ehgmsdk20**
Changes:

- Add TX_STATE_CHANGED event
- Print message when filter deleted
- Add test code

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -288,8 +290,24 @@ export interface ValueChangedEvent {
   values: ValueChangedEventValues;
 }
 
+export interface TxStateChangedEventStates {
```
Comment (platfowner) [inline src/types.ts:293]: question) do we need this interface? can we just use before and after in TxStateChangedEvent directly?
if necessary, we can define TransactionStates here as well?

File: src/types.ts [hunk 2]
```diff
@@ -249,6 +250,7 @@ export interface ValueChangedEventConfig {
 
 export interface TxStateChangedEventConfig {
   tx_hash: string;
+  timeout: number;
```
Comment (platfowner) [inline src/types.ts:253]: could you add 'unit' in the variable name, e.g. timeout_ms?

File: src/event-manager/event-callback-manager.ts [hunk 3]
```diff
@@ -66,6 +69,7 @@ export default class EventCallbackManager {
   createSubscription(filter: EventFilter, eventCallback?: BlockchainEventCallback,
       errorCallback?: (error: any) => void) {
     const subscription = new Subscription(filter);
+    subscription.on('filterDeleted', (payload) => console.log(`Event filter(id: ${payload.filter_id}) is deleted because of ${payload.reason}`));
```
Comment (platfowner) [inline src/event-manager/event-callback-manager.ts:72]: minor) space `filter (`?

File: src/types.ts (hunk 1 above)
Comment (ehgmsdk20) [reply src/types.ts:293]: I created TxStateChangedEventStates just to implement something similar to the ValueChangedEvent above.
```javascript
export interface ValueChangedEventValues {
  before: any;
  after: any;
}
```
I also think this part is unnecessary.

File: src/types.ts (hunk 2 above)
Comment (ehgmsdk20) [reply src/types.ts:253]: Fixed. The blockchain code will also be modified. Thanks!

File: src/event-manager/event-callback-manager.ts (hunk 3 above)
Comment (ehgmsdk20) [reply src/event-manager/event-callback-manager.ts:72]: Fixed. Thanks!

File: src/types.ts (hunk 1 above)
Comment (ehgmsdk20) [reply src/types.ts:293]: Fixed, thanks!

**Comment by ehgmsdk20** [comment]
Thanks for review. Edit done!
PTAL @platfowner

File: src/types.ts (hunk 1 above)
Comment (platfowner) [reply src/types.ts]: 👍 
could you change not to use ValueChangedEventValues either for me? thanks in advance!

**Review by platfowner — APPROVED** [review]
LGTM with some comments.
Nice job done, thanks!

File: src/types.ts (hunk 1 above)
Comment (ehgmsdk20) [reply src/types.ts]: Fixed!

**Comment by ehgmsdk20** [comment]
Thanks for review! Modifications have been completed, but when the blockchain code review is finished, they will be merged together.

File: src/types.ts [hunk 4]
```diff
@@ -273,23 +275,44 @@ export interface ValueChangedEventAuth {
...
 export interface ValueChangedEvent {
   filter_path: string;
   matched_path: string;
   params: any;
   transaction: Transaction;
   event_source: ValueChangedEventSource;
   auth: ValueChangedEventAuth;
-  values: ValueChangedEventValues;
+  values: {
+    before: any;
```
Comment (platfowner) [inline src/types.ts:286]: 👍

**Review by platfowner — APPROVED** [review]
LGTM
Thanks!

**Comment by ehgmsdk20** [comment]
Thanks for your reviews!

**Pull request merged**